samples/apps/copilot-chat-app/webapi/CopilotChat/Models/DocumentMessageContent.cs

```diff
@@ -31,6 +65,54 @@ public override string ToString()
         return JsonSerializer.Serialize(this);
     }
 
+    /// <summary>
+    /// Serialize the object to a formatted string.
+    /// Only successful uploads will be included in the formatted string.
+    /// </summary>
+    /// <returns>A formatted string</returns>
+    public string ToFormattedString()
+    {
+        if (this.Documents.Count == 0)
+        {
+            return string.Empty;
+        }
+
+        var formattedStrings = this.Documents
+            .FindAll(document => document.Status)
+            .Select(document => $"[Name: {document.Name}, Size: {document.Size}]").ToList();
+
+        if (formattedStrings.Count == 1)
+        {
+            return $"Uploaded a document {formattedStrings.First()}.";
+        }
+
+        return $"Uploaded documents: {string.Join(", ", formattedStrings)}.";
+    }
+
+    /// <summary>
+    /// Serialize the object to a formatted string that only
+    /// contains document names separated by comma.
+    /// </summary>
+    /// <returns>A formatted string</returns>
+    public string ToFormattedStringNamesOnly()
+    {
+        if (this.Documents.Count == 0)
+        {
+            return string.Empty;
+        }
+
+        var formattedStrings = this.Documents
+            .FindAll(document => document.Status)
+            .Select(document => $"{document.Name}").ToList();
```

Comment:
Not sure we need the interpolation here?

```suggestion
            .Select(document => document.Name).ToList();
```